config/releases.exs

```diff
@@ -32,26 +32,34 @@ defmodule Environment do
     end
   end
 
-  def get_endpoint_static_url_config(nil), do: nil
-  def get_endpoint_static_url_config(""), do: nil
+  def get_endpoint_url_config(nil), do: nil
+  def get_endpoint_url_config(""), do: nil
 
-  def get_endpoint_static_url_config(host) do
+  def get_endpoint_url_config(uri) do
     [
-      host: host,
-      scheme: get("STATIC_URL_SCHEME"),
-      port: get("STATIC_URL_PORT")
+      host: uri.host,
+      scheme: uri.scheme,
+      port: uri.port
     ]
   end
+
+  def get_uri_part(nil, _), do: nil
+  def get_uri_part("", _), do: nil
+  def get_uri_part(uri, :port), do: uri.port
+  def get_uri_part(uri, :scheme), do: uri.scheme
+  def get_uri_part(uri, :host), do: uri.host
```

Comment:
Would this makes sense?

```suggestion
  def get_uri_part(uri, part), do: Map.get(url, part)
```

Reply:
In fact we simply could do:

```elixir
def get_uri_part(uri, part) when is_struct(uri, URI), do: Map.get(uri, part)
def get_uri_part(_, _), do: nil
```

or

```elixir
def get_uri_part(%URI{host: host}, :host), do: host
def get_uri_part(%URI{port: port}, :port), do: port
def get_uri_part(%URI{scheme: scheme}, :scheme), do: scheme
def get_uri_part(_, _), do: nil
```

What do you think?

Reply:
I really like the last approach! I will try it.